app-backend/backsplash/src/main/scala/services/MosaicService.scala

```diff
@@ -21,61 +31,44 @@ import org.http4s.dsl._
 
 class MosaicService(
     interpreter: BufferingInterpreter = BufferingInterpreter.DEFAULT
-)(implicit t: Timer[IO])
+)(implicit timer: Timer[IO])
     extends Http4sDsl[IO]
-    with RollbarNotifier
-    with Authentication {
+    with RollbarNotifier {
 
   implicit val xa = RFTransactor.xa
 
   // final val eval = MamlTms.curried(RasterVar("identity"), interpreter)
   final val eval = MamlTms.identity[ProjectNode](interpreter)
 
-  object TokenQueryParamMatcher
-      extends QueryParamDecoderMatcher[String]("token")
   object RedBandOptionalQueryParamMatcher
       extends OptionalQueryParamDecoderMatcher[Int]("redBand")
   object GreenBandOptionalQueryParamMatcher
       extends OptionalQueryParamDecoderMatcher[Int]("greenBand")
   object BlueBandOptionalQueryParamMatcher
       extends OptionalQueryParamDecoderMatcher[Int]("blueBand")
+  object TagOptionalQueryParamMatcher
+      extends OptionalQueryParamDecoderMatcher[String]("tag")
 
-  val service: HttpService[IO] =
-    HttpService {
-      case GET -> Root / UUIDWrapper(projectId) / IntVar(z) / IntVar(x) / IntVar(
-            y)
-            :? TokenQueryParamMatcher(token)
+  val service: AuthedService[User, IO] =
+    AuthedService {
+      case req @ GET -> Root / UUIDWrapper(projectId) / IntVar(z) / IntVar(x) / IntVar(
+            y) / _
             :? RedBandOptionalQueryParamMatcher(redOverride)
             :? GreenBandOptionalQueryParamMatcher(greenOverride)
-            :? BlueBandOptionalQueryParamMatcher(blueOverride) =>
-        val authIO: EitherT[IO, Throwable, Boolean] =
-          for {
-            user <- (
-              verifyJWT(token) match {
-                case Right((_, jwtClaims)) =>
-                  EitherT.liftF(
-                    UserDao
-                      .unsafeGetUserById(jwtClaims.getStringClaim("sub"))
-                      .transact(xa)
-                  )
-                case Left(e) =>
-                  EitherT.leftT[IO, User](
-                    new Exception("Failed to validate user from JWT")
-                  )
-              }
-            )
-            authorized <- EitherT.liftF(
-              ProjectDao
-                .authorized(user,
-                            ObjectType.Project,
-                            projectId,
-                            ActionType.View)
-                .transact(xa)
-            )
-          } yield authorized
+            :? BlueBandOptionalQueryParamMatcher(blueOverride)
+            :? TagOptionalQueryParamMatcher(tag) as user =>
+        val authorizationIO =
+          ProjectDao
+            .authorized(user, ObjectType.Project, projectId, ActionType.View)
+            .transact(xa) map { authResult =>
+            if (!authResult)
+              throw NotAuthorized(
+                s"User ${user.id} not authorized to view project $projectId")
```

Comment:
Transformation from exceptions -> responses happened either in the typeclass or the trait.

We can't use the [typeclass](https://github.com/raster-foundry/raster-foundry/blob/feature/js/just-the-non-upgrade-parts-of-error-handling/app-backend/backsplash/src/main/scala/error/ErrorHandling.scala#L23-L43) until upgrades happen, but I could restore the trait to get those transformations back.

Reply:
I put the trait back for now so that we can map these error messages into responses -- as it stands currently, a `NotAuthorized` gets sent to rollbar, but we return a generic error message to the user. I thought it might be useful for us to be able to see what users are without authorization trying to access which projects.